src/Compilers/Core/Portable/PEWriter/MetadataWriter.cs

```diff
@@ -3026,6 +3026,7 @@ private UserStringHandle ReserveUserString(int length, out Blob fixup)
 
         internal const uint LiteralMethodDefinitionToken = 0x80000000;
         internal const uint LiteralGreatestMethodDefinitionToken = 0x40000000;
+        internal const uint SourceDocumentIndex = 0x20000000;
```

Comment:
It uses a bit one to the right of the number above. There are 8 bits to play with, which is likely more than will ever be needed, so using multi-bit combinations probably isn't necessary. If it bothers you, I'm happy to use any bit pattern you think more appropriate.